### PR TITLE
meta: refine error code for package `meta`

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

Not all the errors in meta package have error code.

### What is changed and how it works?

add them.

### Check List 

Tests 

 - None

Code changes

 - None

Side effects

 - None

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: meta/meta.go [hunk 1]
```diff
@@ -67,15 +67,14 @@ var (
 )
 
 var (
-
 	// ErrDBExists is the error for db exists.
-	ErrDBExists = terror.ClassMeta.New(codeDatabaseExists, "database already exists")
+	ErrDBExists = terror.ClassMeta.New(mysql.ErrDBCreateExists, mysql.MySQLErrName[mysql.ErrDBCreateExists])
 	// ErrDBNotExists is the error for db not exists.
-	ErrDBNotExists = terror.ClassMeta.New(codeDatabaseNotExists, "database doesn't exist")
+	ErrDBNotExists = terror.ClassMeta.New(mysql.ErrBadDB, mysql.MySQLErrName[mysql.ErrBadDB])
```
Comment (crazycs520) [inline meta/meta.go]: the message of `mysql.MySQLErrName[mysql.ErrBadDB]` is `"Unknown database '%-.192s'"`, as you can see, it need paramatter, but we use `ErrDBNotExists ` won't have parameter.

File: meta/meta.go [hunk 2]
```diff
@@ -67,15 +67,14 @@ var (
...
 	// ErrTableExists is the error for table exists.
-	ErrTableExists = terror.ClassMeta.New(codeTableExists, "table already exists")
+	ErrTableExists = terror.ClassMeta.New(mysql.ErrNoSuchTable, mysql.MySQLErrName[mysql.ErrNoSuchTable])
 	// ErrTableNotExists is the error for table not exists.
-	ErrTableNotExists = terror.ClassMeta.New(codeTableNotExists, "table doesn't exist")
+	ErrTableNotExists = terror.ClassMeta.New(mysql.ErrTableExists, mysql.MySQLErrName[mysql.ErrTableExists])
```
Comment (zimulala) [inline meta/meta.go]:
```suggestion
	ErrTableNotExists = terror.ClassMeta.New(mysql.ErrNoSuchTable, mysql.MySQLErrName[mysql.ErrNoSuchTable])
```

File: meta/meta.go [hunk 3]
```diff
@@ -67,15 +67,14 @@ var (
...
-
 	// ErrDBExists is the error for db exists.
-	ErrDBExists = terror.ClassMeta.New(codeDatabaseExists, "database already exists")
+	ErrDBExists = terror.ClassMeta.New(mysql.ErrDBCreateExists, mysql.MySQLErrName[mysql.ErrDBCreateExists])
 	// ErrDBNotExists is the error for db not exists.
-	ErrDBNotExists = terror.ClassMeta.New(codeDatabaseNotExists, "database doesn't exist")
+	ErrDBNotExists = terror.ClassMeta.New(mysql.ErrBadDB, mysql.MySQLErrName[mysql.ErrBadDB])
 	// ErrTableExists is the error for table exists.
-	ErrTableExists = terror.ClassMeta.New(codeTableExists, "table already exists")
+	ErrTableExists = terror.ClassMeta.New(mysql.ErrNoSuchTable, mysql.MySQLErrName[mysql.ErrNoSuchTable])
```
Comment (zimulala) [inline meta/meta.go]:
```suggestion
	ErrTableExists = terror.ClassMeta.New(mysql.ErrTableExists, mysql.MySQLErrName[mysql.ErrTableExists])
```

File: meta/meta_test.go [hunk 4]
```diff
@@ -97,6 +97,7 @@ func (s *testSuite) TestMeta(c *C) {
 
 	err = t.CreateDatabase(dbInfo)
 	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[meta:1007]database already exists")
```
Comment (zimulala) [inline meta/meta_test.go]:
```suggestion
	c.Assert(ErrDBExists.Equal(err), IsTrue)
```

File: domain/domain_test.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ package domain
 import (
 	"context"
 	"crypto/tls"
+	"github.com/pingcap/tidb/meta"
```
Comment (zimulala) [inline domain/domain_test.go]: Format it.

File: meta/meta_test.go [hunk 6]
```diff
@@ -290,6 +294,7 @@ func (s *testSuite) TestSnapshot(c *C) {
 	c.Assert(n, Equals, int64(1))
 	_, err = snapMeta.GenGlobalID()
 	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "[structure:5]write on snapshot")
```
Comment (Deardrops) [inline meta/meta_test.go]: This is an unexported error, so we check it by compare string.

**Pull request merged**